tracing/tracing.cc

```diff
@@ -130,7 +130,7 @@ trace_state_ptr tracing::create_session(trace_type type, trace_state_props_set p
         props.set_if<trace_state_props::ignore_events>(!props.contains<trace_state_props::full_tracing>() && ignore_trace_events_enabled());
 
         ++_active_sessions;
-        return make_lw_shared<trace_state>(type, props);
+        return make_lw_shared<opentelemetry_state>(make_lw_shared<trace_state>(type, props), opentelemetry_tracing);
```

Comment:
In line `125` of this file (cannot add a comment at that site), there is this if:

```cpp
if (!may_create_new_session()) {
    return nullptr;
}
```

which will cause OpenTelemetry tracing to be erroneously disabled.